src/include/main/database.h

```diff
@@ -5,11 +5,13 @@
 
 #include "common/api.h"
 #include "common/constants.h"
+#include "common/file_utils.h"
```

Comment:
I don't think you need to include `file_utils.h` in header file. Be careful when adding includes to header files, especially under `common` and `main`, the former one is included in lots of places, and the latter one exposed to clients.
Can you double check if you can remove the include? instead, add forward declaration here https://github.com/kuzudb/kuzu/blob/f05d34893d54b4d75708087a844e6c356907ac84/src/include/main/kuzu_fwd.h#L32